include/flang/semantics/tools.h

```diff
@@ -48,6 +48,8 @@ const DeclTypeSpec *FindParentTypeSpec(const DerivedTypeSpec &);
 const DeclTypeSpec *FindParentTypeSpec(const DeclTypeSpec &);
 const DeclTypeSpec *FindParentTypeSpec(const Scope &);
 const DeclTypeSpec *FindParentTypeSpec(const Symbol &);
+ 
+const DerivedTypeSpec *GetDerivedTypeSpec(const Symbol &);
```

Comment:
Thanks, Peter, I'll add a comment.

Reply:
Actually, adding a comment doesn't seem like the right thing.  If this interface doesn't work for a Symbol that's a type, I should just move it to a place that's not so public and check for its restricted usage.  So that's what I did.